src/rp2_common/pico_lwip/tools/makefsdata.py

```diff
@@ -103,8 +81,12 @@ def process_file_list(fd, input):
 
         # make a variable name
         var_name = str(file.relative_to(input_dir))
-        var_name = var_name.replace(".", "_")
-        var_name = var_name.replace("/", "_")
+        var_name = re.sub(r"\W+", "_", var_name, flags=re.ASCII)
+
+        # Add a suffix if the variable name is used already
+        if any(d["data_var"] == f"data_{var_name}" for d in data):
+            var_name += f"_{len(data)}"
```

Comment:
I guess using `len(data)` as the suffix means that if the first and 6th items are named identically, then the "first duplicate name" would get a suffix of `_5`, which may be unexpected?

An alternative approach might be to add a function like:
```python3
unique_names = set()
def get_unique_name(name):
    if name in unique_names:
        counter = 1
        new_name = f"{name}_{counter}"
        while new_name in unique_names:
            counter += 1
            new_name = f"{name}_{counter}"
        name = new_name
    unique_names.add(name)
    return name
```
and then you could just do `var_name = get_unique_name(var_name)` and then any duplicates would just be numbered sequentially (starting at 1).

But of course this is your PR, so it's up to you / @peterharperuk if you decide to incorporate this suggestion :slightly_smiling_face:

Reply:
I expected this comment 😄 

It just seemed simpler to add those two lines instead of a loop like this. Yes, the numbers will not be sequential, but there _is_ a method to the madness; and the generated code will likely not be scrutinized by users. I couldn't find a technical reason to not do it this way personally; @peterharperuk let me know your thoughts too.

Reply:
> and the generated code will likely not be scrutinized by users

Yeah, I'm possibly getting carried away and losing sight of where the output of all of this **actually** ends up getting used :laughing:

Reply:
Let's keep it as it is.